### ISSUE TITLE
rcl_get_localhost_only function not returning a rmw_localhost_only_t value

### ISSUE THREAD

**Issue by Blast545** [ros2/rcl#669]
While adding tests for the `rcl` package I realized the `localhost.c` function `rcl_get_localhost_only` uses this logic to return the possible value of the localhost_only variable:

https://github.com/ros2/rcl/blob/e47042cba4f32f77b361d0f87d905b9386cadde8/rcl/src/rcl/localhost.c#L44

Which only returns `0` or `1`, but the current possible values for this variable are `0-1-2`, as implemented in [rmw/localhost.h](https://github.com/ros2/rmw/blob/eaf5932723efa0be3fda4847b69e09d3753f0a6b/rmw/include/rmw/localhost.h#L26):

```
typedef enum RMW_PUBLIC_TYPE rmw_localhost_only_t
{
  /// Uses ROS_LOCALHOST_ONLY environment variable.
  RMW_LOCALHOST_ONLY_DEFAULT = 0,
  /// Forces using only localhost.
  RMW_LOCALHOST_ONLY_ENABLED = 1,
  /// Forces disabling localhost only.
  RMW_LOCALHOST_ONLY_DISABLED = 2,
} rmw_localhost_only_t;
```

**Comment by Blast545** [comment]
After checking the expected usage of this function, it is expected that the result of this function to be a boolean value because the result coming from this is then fed to `rmw_create_node()` that takes `localhost_only` value as a boolean.

I think the solution to this would be changing the `rcl_get_localhost_only` signature to return boolean instead.

**Comment by Blast545** [comment]
Closed with #670
